Package/CreatePackage/Package.cs

```diff
@@ -673,10 +680,57 @@ private static void UpdateVersionInfo(string tempDir, List<PackageFile> files, S
             log.Info(timer,"Updated assembly version info using Mono method.");
         }
 
+        private static bool VerifyArchiveIntegrity(FileStream fs)
+        {
+            log.Debug($"Verifying package integrity.");
+            // Ensure the stream is fully written to the disk
+            fs.Flush();
+
+            {   // Verify that we can read the package definition
+                try
+                {
+                    PackageDef.FromPackage(fs.Name);
+                }
+                catch (Exception ex)
+                {
+                    log.Error($"Error verifying package integrity.");
+                    log.Debug(ex);
+                    return false;
+                }
+            }
+
+            // Rewind the stream
+            fs.Seek(0, SeekOrigin.Begin);
+            {   // Verify that we can extract the archive without errors
+                using var ms = new MemoryStream();
+                using var zip = new ZipArchive(fs, ZipArchiveMode.Read, true);
+                foreach (var part in zip.Entries)
+                {
+                    // Rewind the stream to overwrite the previous file on each iteration
+                    ms.Seek(0, SeekOrigin.Begin);
+                    try
+                    {
+                        // Read the stream to the end to verify it can be extracted
+                        part.Open().CopyTo(ms);
+                    }
+                    catch (InvalidDataException)
+                    {
+                        log.Error($"Error verifying package integrity. " +
+                                  $"Archive entry '{part.FullName}' is corrupted. " +
+                                  $"Please retry the package creation. " +
+                                  $"If the error persists, consider creating an issue.");
+                        return false;
+                    }
+                }
+            }
+
+            return true;
```

Comment:
maybe here add log.Debug(timer, "Package verified.");?

The timer is so that we can keep track of how long time we spend on this.